src/messaging/ReliableMessageContext.cpp

```diff
@@ -136,6 +136,7 @@ CHIP_ERROR ReliableMessageContext::FlushAcks()
 #if !defined(NDEBUG)
             ChipLogDetail(ExchangeManager, "Flushed pending ack for MsgId:%08" PRIX32, mPendingPeerAckId);
 #endif
+            SetAckPending(false);
```

Comment:
if piggy back message does not do the SetAckPending(false), then a standalong ack message will be sent in the following MRP tick, right?

We have another issue since those flags might get manipulated from multiple threads,  but we already have a separate thread to discuss the general racing issue.

Reply:
> if piggy back message does not do the SetAckPending(false),

But it does, right?  That call is in `ExchangeMessageDispatch::SendMessage` where we piggyback.

So to be clearer: we should have `SetAckPending(false)` in piggyback where we have it now and in `SendStandaloneAckMessage`, which are the two places that can send acks.  And we should not need it anywhere else.  That means not in `FlushAcks` and not in `ReliableMessageMgr::ExecuteActions`.